Feature/aql subquery execute parallel gather 2 optimize unsorted gather

markuspf opened the pull request:
Team gordo internal

goedderz commented on arangod/Aql/UnsortedGatherExecutor.cpp:
```diff
@@ -79,102 +79,45 @@ auto UnsortedGatherExecutor::produceRows(typename Fetcher::DataRange& input,
-    return {input.upstreamState(currentDependency()), Stats{}, call.getSkipCount(), callSet};
-  }
-}
+  while (call.needSkipMore()) {
```
If I'm not mistaken, this `while` loop always ends in a `return`, and after it sits a `TRI_ASSERT(false)`. So maybe
remove the loop and
```suggestion
  TRI_ASSERT(call.needSkipMore());
```
here instead?

markuspf commented:
I updated `skipRowsRange` and removd the while loop, relying on the `ExecutionBlockImpl` state machine to loop; I also added the `fullCount` case where we call `skipAllForDependency` instead of `skipForDependency` because we have no count in hand.
